pkg/sql/catalog/bootstrap/previous_release.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package bootstrap
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/pkg/config/zonepb"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/errors"
+)
+
+// InitialValuesForTenantV222 returns the initial values as produced by the
+// binary for the 22.2 release, for a non-system tenant. The only difference
+// is that the contents of the system.zones table are overwritten with the new
+// values.
+func InitialValuesForTenantV222(
+	codec keys.SQLCodec,
+	defaultZoneConfig *zonepb.ZoneConfig,
+	defaultSystemZoneConfig *zonepb.ZoneConfig,
+) (kvs []roachpb.KeyValue, splits []roachpb.RKey, _ error) {
+	h := sha256.Sum256([]byte(nonSystem222))
+	if actualHash := hex.EncodeToString(h[:]); actualHash != nonSystem222SHA256 {
+		return nil, nil, errors.AssertionFailedf("expected %s, got %s", nonSystem222SHA256, actualHash)
+	}
+	var initialKVs []roachpb.KeyValue
+	initialKVs, splits, err := InitialValuesFromString(codec, nonSystem222)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Replace system.zones entries.
+	kvs = InitialZoneConfigKVs(codec, defaultZoneConfig, defaultSystemZoneConfig)
+	zonesTablePrefix := codec.TablePrefix(keys.ZonesTableID)
+	for _, kv := range initialKVs {
+		if !bytes.Equal(zonesTablePrefix, kv.Key[:len(zonesTablePrefix)]) {
+			kvs = append(kvs, kv)
+		}
+	}
+	sort.Sort(roachpb.KeyValueByKey(kvs))
+	return kvs, splits, nil
+}
+
+const (
+	// nonSystem222 is the hard-coded string representation of the
+	// initial values for a non-system tenant as produced by the 22.2
+	// release binary.
+	//
+	// This string literal was generated by copying over the expected results
+	// for the TestInitialValuesToString test in the release-22.2 branch.
+	nonSystem222 = `[{"key":""}
```

Comment:
what's this first entry about?

Reply:
Keys without values are split keys and this one is the split we define at the tenant prefix, but in this datum the prefix is stripped away leaving the empty string.